### PR TITLE
Using Glad2 for new Extensions

### PR DESCRIPTION
**Pull request by gralkapk**
Replaced glad with glad2 including symbols for new extensions.

## Test Instructions

Has not reached all plugins.
Respective maintainers should test it.

### CODE REVIEW

File: frontend/resources/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +38,7 @@ target_include_directories(${PROJECT_NAME} PUBLIC
   #$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
   #$<INSTALL_INTERFACE:include/>
 )
-target_link_libraries(${PROJECT_NAME} PUBLIC json glm)
+target_link_libraries(${PROJECT_NAME} PUBLIC json glm glad)
```
Comment (geringsj) [inline frontend/resources/CMakeLists.txt]: Resources should not need to link against libraries

File: frontend/resources/src/OpenGL_Context.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        return false;
+    }
+    return true;
+}
+
+
+bool megamol::frontend_resources::OpenGL_Context::isExtAvailable(std::string const& ext) const {
+    auto const fit = std::find(ext_.begin(), ext_.end(), ext);
+    return fit != ext_.end();
+}
```
Comment (geringsj) [inline frontend/resources/src/OpenGL_Context.cpp]: Move whole OpenGL_Context.cpp to `frontend/services/opengl_glfw/`, the GL Service already links against glad and if there is no GL there is no `OpenGL_Context`

Comment (geringsj) [reply frontend/resources/src/OpenGL_Context.cpp]: Or make OpenGL_Context.h/OpenGL_Context.cpp not depend on glad by hiding relevant implementation details in the GL Service

File: frontend/resources/src/OpenGL_Context.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+bool megamol::frontend_resources::OpenGL_Context::areExtAvailable(std::string const& exts) const {
+    std::istringstream iss(exts);
+    bool avail = true;
+    std::for_each(std::istream_iterator<std::string>(iss), std::istream_iterator<std::string>(),
+        [&avail, this](auto const& ext) { avail = avail && isExtAvailable(ext); });
+    return avail;
+}
```
Comment (geringsj) [inline frontend/resources/src/OpenGL_Context.cpp]: Move whole OpenGL_Context.cpp to frontend/services/opengl_glfw/ so we can avoid linking against glad in the resources CMakeLists

**Review by moritz-h — DISMISSED** [review]

**Review by moritz-h — DISMISSED** [review]

**Review by moritz-h — APPROVED** [review]

**Pull request merged**